File: Kth-Missing-Positive-Number.py
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        # l=1
        # h=max(arr)
        # r=[]
        # for i in range(l,h+k+1):
        #     if i in arr:
        #         continue
        #     else:
        #         r.append(i)
        # # if len(r)>=k:
        # #     return r[k-1]
        # ans=r[k-1]
        # return ans 
        l=0
        h=len(arr)-1
        while(l<=h):
            mid=(l+h)//2
            missingArray=arr[mid]-(mid+1)
            if missingArray<k:
                l=mid+1
            else:
                h=mid-1
        return l+k
         
        

File: Kth-Missing-Positive-Number.js
/**
 * @param {number[]} arr
 * @param {number} k
 * @return {number}
 */
var findKthPositive = function(arr, k) {
    // let l=1
    // let h=Math.max(...arr)
    // let r=[]
    // for (let i=l ;i<=h+k;i++) {
    //     if ( arr.includes(i)){
    //         continue
    //     }else{
    //         r.push(i)
    //     }
    // }
    // let ans=r[k-1]
    // return ans

    let l=0
    let h= arr.length - 1
    while(l<=h){
        let mid=Math.floor((l+h)/2)
        let missingArray=arr[mid]-(mid+1)
        if (missingArray<k){
            l=mid+1
        }
        else{
            h=mid-1
        }
    }
    return l+k
};
